server/container/guice/protocols/jmap/src/main/java/org/apache/james/jmap/draft/JMAPConfigurationStartUpCheck.java

```diff
@@ -69,8 +105,71 @@ private CheckResult checkSecurityKey() {
         }
     }
 
+    private void loadSecurityKey() throws Exception {
+        try {
+            securityKeyLoader.load();
+        } catch (FileNotFoundException e) {
+            if (runArguments.contain(Argument.GENERATE_KEYSTORE)) {
+                LOGGER.warn("Can not load asymmetric key from configuration file" + e.getMessage());
+                LOGGER.warn("James will try the auto-generate an asymmetric key. It is just for development, should not use it on production");
+                generateKeystore();
+            } else {
+                throw e;
+            }
+        }
+    }
+
     @Override
     public String checkName() {
         return CHECK_NAME;
     }
+
+    private void generateKeystore() throws Exception {
```

Comment:
It is not the JMAP startUp checks that should generate the keystore...

Reply:
It is based on your suggestion "please move the key generation out of JMAP-draft. Suggestion: server/container/guice/common"

Can you give more detail?  
I tried to move the SecurityKeyload to `server/container/guice/common`, but see it is a bit not correct, So I only move the generateKeystore method

Reply:
I can try to get a look at it ^^